### DISCUSSION TITLE
add ability to enable / disable consumer credentials without deleting them

### DISCUSSION THREAD

**Discussion by ahmadnassri** [Kong/kong#7202] [Ideas and feature requests]
scenario:
- I want to disable certain consumers from using certain credentials temporarily without changing their keys.

expectation:
- I can change `enabled` flag on a credentials object (in `oauth2` or other authentication plugins)

**Comment by ahmadnassri** [comment]
cc: @lucamaraschi

**Comment by b6t** [comment]
+1 or at least the ability to delete a key. If a key is compromised there are no options to invalidate the compromised key and issue a new one. is there?

**Comment by subnetmarco** [comment]
@coreybeaumont yes, you can execute an HTTP DELETE request to that key.

**Comment by b6t** [comment]
@thefosk thanks - the structure of the url was not initially apparent to successfully delete. Anyways, active | inactive would still be a great help.

**Comment by rzschech** [comment]
I would like this too. I would do the following:

1. user signs up with email address and password
2. create consumer and basic auth credentials where username=email address, password=password and disabled=true
3. send email address verification email to user
4. user can't login with basic auth credentials
5. user clicks link in verification email
6. patch basic auth credential disabled=false

This allows me to store the password while the email address verification is in progress.

**Comment by parajanoff** [comment]
+1

**Comment by ahmedbrandver** [comment]
+ 1
